vcs/internal/script.go

```diff
@@ -0,0 +1,53 @@
+package internal
+
+import (
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"time"
+)
+
+var gen rand.Source
+
+func init() {
+	gen = rand.NewSource(time.Now().UnixNano())
+}
+
+// Constructs platform-specific temporary script file with a given prefix
+// On Windows such a file must have .bat extension
+func ScriptFile(prefix string) (string, error) {
+
+	if runtime.GOOS == "windows" {
+		for {
+			tempFile := filepath.Join(os.TempDir(), prefix+strconv.FormatInt(gen.Int63(), 36)+".bat")
```

Comment:
Can we just use `os.TempFile()` here and forgo the need to generate the random number ourselves?